### PR TITLE
Add instructions on how to load translations with webpack

### PR DESCRIPTION
**Pull request by brenoc**
I use webpack in my front-end projects and the translations are not served on the back-end. They are static json files inside the front-end application.

I managed to use this plugin with webpack loading the translations files async.

I think this solution could be shared with the community, since I had some difficulties finding out how should I write my custom xhr function and its callback and, later, with the parse function. I thought that will help someone else doing those same mistakes.

### CODE REVIEW

**Comment by jamuhl** [comment]
not sure...what's the benefit of using the bundle-loader instead of the xhr function built in?

would it make sense to have a separate backend just for webpack usage? so devs just can use that without manual override xhr and parse?

**Comment by brenoc** [comment]
@jamuhl 

Most front end applications using webpack are build this way:

```
src/
dist/
webpack.config.js
```

Webpack get the files in `src` process them (transpile with babel, convert jsons to js files, uglify, etc) and dump them in `dist`. For webpack to work on the `src` files you need to require them using `import x from './x';` or `var x = require('./x');`. Those files will be **bundled together** in a single file (also called an entry-point).

When you want to require files asynchronously using webpack you may want to use bundle-loader. The bundle loader will include the required files in the webpack build, but will not bundle it together with the entry-point, it will create a separate file, since this file will be required asynchronously.

```js
let waitForLocale = require('bundle!./locales/'+url+'.json');
```
Webpack will interpret this line and all the files in `./locales/*.json` will be included in the build like so:

```
src/
-- app.js
-- locales/
---- en.json
---- es.json
dist/
-- app.js
-- 1.app.js (this was formely en.json)
-- 2.app.js (this was formely es.json)
```

**TL;DR:** Using the built in xhr function, will not work here, because the translation files will not be included in the `dist/` folder, since no file processed by webpack required those files.

**Comment by jamuhl** [comment]
sorry was not clear enough...i know the build process of webpack...it just wasn't clear to me you want the locales bundled to dist.

We keep our translation out of the build main reason is we update translations without deploying a new version.

Will merge this. But still would love to see a `i18next-webpack-backend` ;)

**Comment by brenoc** [comment]
Oh, now I get it what you mean by a new backend, you mean a new i18next plugin! Yes, that could be a good solution as well!

Maybe I'll revisit and do this, but it's up for grabs anyways.

**Comment by jamuhl** [comment]
yes...meant a complete backend plugin...so one just needs to grab that and no manual override in the basic xhr-backend.

would be great to see this...right now i'm hard on time...so if you get on this before me feel free to add it to documentation: https://github.com/i18next/i18next.com/blob/master/pages/docs/ecosystem/index.md
